src/data/fintype/basic.lean

```diff
@@ -434,6 +437,17 @@ lemma fintype.card_le_of_injective [fintype α] [fintype β] (f : α → β)
   (hf : function.injective f) : fintype.card α ≤ fintype.card β :=
 finset.card_le_card_of_inj_on f (λ _ _, finset.mem_univ _) (λ _ _ _ _ h, hf h)
 
+/--
+The pigeonhole principle for finitely many pigeons and pigeonholes.
+-/
+lemma fintype.pigeonhole [fintype α] [fintype β] (f : α → β) (h : fintype.card β < fintype.card α) :
+  ∃ x y, x ≠ y ∧ f x = f y :=
+begin
+  classical, by_contra hf, push_neg at hf,
+  refine nat.lt_le_antisymm h (fintype.card_le_of_injective f _),
+  intros x y, contrapose, apply hf,
+end
```

Comment:
Can this be written in terms of `pigeonhole`, or vice versa?

Might be nice to have the doc-comments reference the other lemma.

Reply:
Yeah, it's possible to write `fintype.pigeonhole` in terms of `finset.pigeonhole`:
```lean
lemma fintype.pigeonhole' [fintype α] [fintype β] (f : α → β) (h : fintype.card β < fintype.card α) :
  ∃ x y, x ≠ y ∧ f x = f y :=
by { obtain ⟨x, _, y, _, h⟩ := @finset.pigeonhole α β univ univ h f (by simp),
     exact ⟨x, y, h⟩ }
```
I just figured it was a trivial application of `fintype.card_le_of_injective`, so it doesn't hurt to have independent proofs, but I might change it.

I agree it would be nice to have doc comments -- I'll be sure to add them once it's clear what these lemmas should be called and where they should go.

Reply:
Changed it and added some documentation